tests/languages/python_test.py

```diff
@@ -44,6 +44,10 @@ def test_sys_executable_matches_does_not_match(v):
     ),
 )
 def test_find_by_sys_executable(exe, realpath, expected):
+    def mocked_find_executable(exe):
+        return exe.rpartition('/')[2]
     with mock.patch.object(sys, 'executable', exe):
         with mock.patch('os.path.realpath', return_value=realpath):
-            assert python._find_by_sys_executable() == expected
+            with mock.patch('pre_commit.parse_shebang.find_executable',
```

Comment:
please use `mock.patch.object` instead, `mock.patch` has import side-effects and can lead to some nasty patch leaking issues.  `mock.patch.object` is also more explicit about the target

also, that feel when you don't run the pre-commit hooks while contributing to pre-commit :laughing: